### PR TITLE
Better local docker instructions for misbehaving-jmx-server

### PR DESCRIPTION
**Pull request by scottopell**
This PR adds a full docker-compose that includes the Agent as well as the jmx testing server with AD labels.

By running both misbehaving-jmx-server and the agent in the same bridge network (default for docker-compose), we're able to use the docker-compose hostname to connect from the Agent.

Other options for docker development will require a manual second step where the rmi-hostname is set on the misbehaving-jmx-server supervisor.

The above applies to a scenario where both the agent and misbehaving-jmx-server are running in a container. If you are running only the misbehaving-jmx-server in a container and the agent on the host, networking can be a bit tricky, stay tuned for a future PR documenting how to do this.

### CODE REVIEW

**Review by gh123man — DISMISSED** [review]

File: tools/misbehaving-jmx-server/README.md [hunk 1]
```diff
@@ -52,9 +52,18 @@ There are a couple of ways you can get the Agent to pull metrics from this test
 Copy `misbehaving-jmxfetch-conf.yaml` to `/etc/datadog-agent/conf.d/` and just run the `with-dependencies` jar created by Maven.
 You will need to restart the Agent to pick up the config.
 
+### Using Docker Compose
+
+```shell
+$ docker compose up
+```
+
+The Agent will auto discover the container and begin to collect metrics from it.
```
Comment (drichards-87) [inline tools/misbehaving-jmx-server/README.md:61]:
```suggestion
The Agent will auto-discover the container and begin to collect metrics from it.
```

File: tools/misbehaving-jmx-server/README.md [hunk 2]
```diff
@@ -52,9 +52,18 @@ There are a couple of ways you can get the Agent to pull metrics from this test
...
+
 ### Using Docker
 
-After building the `misbehaving-jmx-server` you can simply run:
+If your container's IP is directly
+accessible by your Agent, you can use the following run command and use AD.
```
Comment (drichards-87) [inline tools/misbehaving-jmx-server/README.md:66]:
```suggestion
accessible by your Agent, you can use the following `run` command and use AD.
```

File: tools/misbehaving-jmx-server/README.md [hunk 3]
```diff
@@ -66,10 +75,7 @@ misbehaving-jmx-server
 
 The Agent will auto discover the container and begin to collect metrics from it.
 
-### Using Docker Compose
+Note that this implicitly sets the `RMI_HOSTNAME` to `localhost` which is where
+the host port mapping comes into play. If this is giving you trouble, consider
+using the docker-compose setup above.
```
Comment (drichards-87) [inline tools/misbehaving-jmx-server/README.md:80]:
```suggestion
using the [docker-compose setup](#using-docker-compose).
```

**Review by drichards-87 — APPROVED** [review]
Left you a couple of minor suggestions from Docs and approved the PR.

**Pull request merged**